benchmark-mockingbird/src/test/java/com/anthonycr/benchmark/BenchmarkTest.kt

```diff
@@ -77,10 +78,10 @@ class BenchmarkTest {
             interface4.doThing2(1, true)
             interface5.doThing("test1", 1, true)
             interface6.doThing("test1", true, 1, 1)
-            interface7.verifyParams({
-                val exception = it[0]
-                exception is Exception && exception.message == "test1"
-            }) { doThing(Exception("test1")) }
+            interface7.verifyParams(
+                func = Interface7::doThing,
+                p0 = eq(Exception("test1")) { it.message == "test1" }
+            )
```

Comment:
I think this is a better experience for verifying parameters. The only downside is requiring that the parameter be provided for all invocations even when we are using this `eq` function and ignoring what we are passing. The reason is that we aren't doing any mocking, so we can't make up a fake parameter to pass and then ignore.

Possibly could improve this by using `Comparable` and encouraging implementations to be created for parameters that can't be directly compared like `Exception`.